File: frontend/src/shared/enums/backend/execution/execution-status-enum.ts
export const ExecutionStatusEnum = {
  RUNNING: "RUNNING",
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
  INCONCLUSIVE: "INCONCLUSIVE",
  STOPPED: "STOPPED",
  ERRORED: "ERRORED",
  ABANDONED: "ABANDONED",
} as const;

export type ExecutionStatusEnum =
  (typeof ExecutionStatusEnum)[keyof typeof ExecutionStatusEnum];

File: backend/Core/Enums/Execution/ExecutionStatusEnum.cs
namespace Core.Enums
{
    public enum ExecutionStatusEnum
    {
        RUNNING,
        COMPLETED,
        FAILED,

        /// <summary>The walk reached the end and no End Execution ever said how it went.</summary>
        INCONCLUSIVE,
        STOPPED,
        ERRORED,

        /// <summary> Left RUNNING by a process that never came back. Set by a sweep at startup. </summary>
        ABANDONED,
    }
}
